WooCommerce/WooCommerceTests/ViewRelated/Blaze/BlazeConfirmPaymentViewModelTests.swift

```diff
@@ -166,11 +214,156 @@ final class BlazeConfirmPaymentViewModelTests: XCTestCase {
+            default:
+                break
+            }
+        }
+
+        await viewModel.updatePaymentInfo()
+        await viewModel.submitCampaign()
+
+        // Then
+        let campaign = try XCTUnwrap(submittedCampaign)
+        XCTAssertEqual(campaign.mainImage.url, media.src)
+        XCTAssertEqual(campaign.mainImage.mimeType, media.mimeType)
+    }
+
+    func test_campaignCreationError_is_correct_when_retrieve_media_fails() async {
+        // Given
+        let viewModel = BlazeConfirmPaymentViewModel(productID: sampleProductID,
+                                                     siteID: sampleSiteID,
+                                                     campaignInfo: .fake(),
+                                                     image: .init(image: .init(), source: .productImage(image: .fake())),
+                                                     stores: stores) {}
+        XCTAssertNil(viewModel.campaignCreationError)
+
+        // When
+        mockRetrieveMedia(with: .failure(NSError(domain: "test", code: 500)))
+        mockCampaignCreation(with: .success(Void()))
+        await viewModel.updatePaymentInfo()
+        await viewModel.submitCampaign()
+
+        // Then
+        XCTAssertEqual(viewModel.campaignCreationError, .failedToFetchCampaignImage)
+    }
+
+    // MARK: Upload image
+
+    func test_campaign_is_submitted_with_uploaded_media_src_and_mimeType() async throws {
+        // Given
+        let viewModel = BlazeConfirmPaymentViewModel(productID: sampleProductID,
+                                                     siteID: sampleSiteID,
+                                                     campaignInfo: .fake(),
+                                                     image: .init(image: .init(), source: .asset(asset: PHAsset())),
+                                                     stores: stores) {}
+
+        // When
+        let media = Media.fake().copy(mimeType: "image/jpeg", src: "http://example.com/test.jpeg")
+        mockUploadMedia(with: .success(media))
+
+        var submittedCampaign: CreateBlazeCampaign?
+        stores.whenReceivingAction(ofType: BlazeAction.self) { action in
+            switch action {
+            case let .fetchPaymentInfo(_, onCompletion):
+                onCompletion(.success(.fake().copy(savedPaymentMethods: [self.samplePaymentMethod])))
+            case let .createCampaign(campaign, _, onCompletion):
+                submittedCampaign = campaign
+                onCompletion(.success(Void()))
+            default:
+                break
+            }
+        }
+
+        await viewModel.updatePaymentInfo()
+        await viewModel.submitCampaign()
+
+        // Then
+        let campaign = try XCTUnwrap(submittedCampaign)
+        XCTAssertEqual(campaign.mainImage.url, media.src)
+        XCTAssertEqual(campaign.mainImage.mimeType, media.mimeType)
+    }
+
+    func test_upload_image_is_retired_one_time_if_upload_fails() async throws {
```

Comment:
Nit:

```suggestion
    func test_upload_image_is_retried_one_time_if_upload_fails() async throws {
```

Reply:
🕵️ Fixed in https://github.com/woocommerce/woocommerce-ios/pull/12006/commits/96bfd7bd52d7a075c199e70cb19efa564bd1bcfa